src/modm/platform/heap/cortex/no_heap.c.in

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2020, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <reent.h>
+#include <modm/architecture/utils.hpp>
+
+modm_weak
+void __modm_initialize_memory(void) {}
+
+// ----------------------------------------------------------------------------
+modm_section("{{ no_heap_section }}") modm_weak
+void* _sbrk_r(struct _reent *r, ptrdiff_t size)
```

Comment:
This is actually perfect, since `_sbrk_r` is not implemented in newlib at all, so I can just implement it here as a `weak` function with just the section attribute and no other linker flags (particularly no function wrapping).

This means the user can supply their own malloc implementation if they do not include the `:platform:heap` module, by either just simply overwriting `_sbrk_r` to use the default newlib allocator or by additionally wrapping the `_malloc_r` etc function via the project collectors settings.